File: 1642-water-bottles/water-bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int sum=0;
        int num_full=numBottles;
        int num_empty=0;
        while(num_full>0)
        {
            sum= sum+num_full;
            num_empty=num_empty+num_full;
            num_full = num_empty/numExchange;
            num_empty=num_empty%numExchange;
        }
        return sum;
    }
};

File: 1642-water-bottles/water-bottles.py
import math
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        sum=0
        num_full=numBottles
        num_empty=0
        while(num_full>0):
            sum+=num_full
            num_empty+=num_full
            num_change = math.floor(num_empty/numExchange)
            num_full=num_change
            num_empty-=num_change*numExchange
        return sum
        